lib/ansible/module_utils/facts/hardware/linux.py

```diff
@@ -483,7 +471,28 @@ def get_mount_facts(self):
 
             mount_info.update(mount_statvfs_info)
 
-            mounts.append(mount_info)
+            return mount_info
+
+    @timeout.timeout()
+    def get_mount_facts(self):
+        mount_facts = {'mounts': []}
+
+        bind_mounts = self._find_bind_mounts()
+        uuids = self._lsblk_uuid()
+        mtab_entries = self._mtab_entries()
+
+        # start threads to query each mount
+        mounts = []
+        async_results = []
+        pool = ThreadPool(processes=min(len(mtab_entries), 8))
```

Comment:
If you're trying to make sure we don't overtax the cpu, use this: ```min(len(mtab_entries), multiprocessing.cpu_count())```

Reply:
done